app/ui/browser-modern/util/record-constructors.js

```diff
@@ -0,0 +1,22 @@
+/*
+Copyright 2016 Mozilla
+
+Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+this file except in compliance with the License. You may obtain a copy of the
+License at http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software distributed
+under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+CONDITIONS OF ANY KIND, either express or implied. See the License for the
+specific language governing permissions and limitations under the License.
+*/
+
+const RECORDS = new Set();
+
+export function register(record) {
+  RECORDS.add(record);
+  return record;
+}
+
+export function getAllRegisteredRecords() {
```

Comment:
100% meh. I'll convert it to an array, but I expect us to be smart enough not to modify the array returned by `getAllRegisteredRecords`. That being said, future versions of ourselves might not be careful, so I'll clone it.